app/controllers/application_controller.rb

```diff
@@ -5,6 +5,110 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if logged_in? then redirect "/tweets" end
+    erb :"/users/create_user"
+  end
+
+  get '/tweets' do
+    if !logged_in? then redirect "/login" end
+    @user=current_user
+    erb :"/tweets/tweets"
+  end
+
+  get '/tweets/new' do
+    if !logged_in? then redirect "/login" end
+    @user=current_user
+    erb :"/tweets/new"
+  end
+
+  get "/tweets/:id" do
+    if !logged_in? then redirect "/login" end
+   @session=session
+    @tweet = Tweet.find(params[:id])
+    erb :"/tweets/show_tweet"
+  end
+
+  get '/tweets/:id/edit' do
+    if !logged_in? then redirect "/login" end
+    @tweet = Tweet.find(params[:id].to_i)
+    @user=@tweet.user
+    if session[:id]!= @user.id then redirect "/tweets" end
+    erb :"/tweets/edit_tweet"
+  end
+
+  post '/tweets/new' do
+    if !logged_in? then redirect "/login" end
+    if params[:content]=="" then redirect "/tweets/new" end
+    @tweet = Tweet.create(content: params[:content], user_id: session[:id])
+    redirect "/tweets"
+  end
+
+  post '/tweets/:id' do
+    if params[:content]=="" then redirect "/tweets/#{params[:id]}/edit" end
+    @tweet=Tweet.find(params[:id])
+    @tweet.update(content: params[:content])
+    @tweet.save
+    redirect "/tweets"
+  end
+
+  post '/tweets/:id/delete' do
+    tweet=Tweet.find(params[:id])
+    if tweet.user.id==session[:id] then tweet.delete end
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.